shell/platform/darwin/macos/framework/Source/FLEViewController.mm

```diff
@@ -612,11 +615,34 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
   if ([call.method isEqualToString:@"SystemNavigator.pop"]) {
     [NSApp terminate:self];
     result(nil);
+  } else if ([call.method isEqualToString:@"Clipboard.getData"]) {
+    result([self getClipboardData:call.arguments]);
+  } else if ([call.method isEqualToString:@"Clipboard.setData"]) {
+    [self setClipboardData:call.arguments];
+    result(nil);
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
+#pragma mark - ClipboardData
+
+- (NSDictionary*)getClipboardData:(NSString*)format {
+  NSPasteboard* pasteboard = [NSPasteboard generalPasteboard];
+  if (!format || [format isEqualToString:@(kTextPlainFormat)]) {
+    id stringInPasteboard = [pasteboard stringForType:NSPasteboardTypeString];
+    // The pasteboard may contain an item but it may not be a string (an image for instance).
+    return stringInPasteboard == nil ? nil : @{@"text" : stringInPasteboard};
+  }
+  return nil;
+}
+
+- (void)setClipboardData:(NSDictionary*)data {
+  NSPasteboard* pasteboard = [NSPasteboard generalPasteboard];
+  [pasteboard declareTypes:@[ NSPasteboardTypeString ] owner:self];
```

Comment:
Isn't this only need for the old style of clipboard handling? I thought the new flow was to `clearContents` and then set contents.